### PR TITLE
Fix stack upgrade to use --keep-dependencies

### LINKED ISSUE #2804: e2e specs: stack upgrade for a stack that has dependencies when a dependency is removed removes the stack
From #2707: looks like `kontena stack upgrade` needs to be fixed to use `--keep-dependencies`, not `--keep-dependent`

```
stack upgrade
  for a stack that has dependencies
    when a dependency is removed

 [done] Parsing kontena.yml     
 [done] Reading stack twemproxy from master     

Stacks to be removed because they are no longer depended on:
- twemproxy-redis_from_yml-monitor

ERROR: Unrecognised option '--keep-dependent'

See: 'kontena_cli.rb stack remove --help'
      removes the stack (FAILED - 1)

Failures:

  1) stack upgrade for a stack that has dependencies when a dependency is removed removes the stack
     Failure/Error: expect(k.code).to eq (0)
     
       expected: 0
            got: 1
     
       (compared using ==)
     # ./spec/features/stack/upgrade_spec.rb:122:in `block (5 levels) in <top (required)>'
     # ./spec/support/shell.rb:33:in `block in with_fixture_dir'
     # ./spec/support/shell.rb:32:in `chdir'
     # ./spec/support/shell.rb:32:in `with_fixture_dir'
     # ./spec/features/stack/upgrade_spec.rb:119:in `block (4 levels) in <top (required)>'
```

### PR DESCRIPTION
**Pull request by kke**
Fixes #2804

(from #2707)

### CODE REVIEW

File: cli/lib/kontena/cli/stacks/upgrade_command.rb [hunk 1]
```diff
@@ -103,8 +103,8 @@ def execute
           puts "#{pastel.red('Warning:')} This can not be undone, data will be lost."
         end
         confirm unless force?
-        removes.reverse_each do |r|
-          Kontena.run!('stack', 'remove', '--force', '--keep-dependent', r)
+        removes.reverse_each do |removed_stack|
+          Kontena.run!('stack', 'remove', '--force', '--keep-dependencies', removed_stack)
           merged.delete(r)
```
Comment (SpComb) [inline cli/lib/kontena/cli/stacks/upgrade_command.rb]:
```
 [error] NameError : undefined local variable or method `r' for #<Kontena::Cli::Stacks::UpgradeCommand:0x005583418f2b08>
         See /root/.kontena/kontena.log or run the command again with environment DEBUG=true set to see the full exception
```

Comment (kke) [reply cli/lib/kontena/cli/stacks/upgrade_command.rb]: Fix

File: cli/lib/kontena/cli/stacks/yaml/stack_file_loader.rb [hunk 2]
```diff
@@ -27,7 +27,9 @@ def self.loaders
       # @return [StackFileLoader]
       def self.for(source, parent = nil)
         loader = loaders.find { |l| l.match?(source, parent) }
-        raise "Can't determine stack file origin for '#{source}'" if loader.nil?
+        if loader.nil?
+          raise RuntimeError, "Not found: no such file #{source} or invalid uri scheme"
```
Comment (SpComb) [inline cli/lib/kontena/cli/stacks/yaml/stack_file_loader.rb]: This slipped in from  #2809?

Comment (kke) [reply cli/lib/kontena/cli/stacks/yaml/stack_file_loader.rb]: Yes

Comment (SpComb) [reply cli/lib/kontena/cli/stacks/yaml/stack_file_loader.rb]: Huh, after merging that, the diff still shows up in the PR, without it being a conflict... let's see what happens.

**Pull request merged**